### PR TITLE
feat: add Hoon language support

### PR DESCRIPTION
**Pull request by urbit-pilled**
Adding a parser for the language [hoon](https://developers.urbit.org/reference/hoon)

### CODE REVIEW

**Comment by amaanq** [comment]
experimental is set in parsers.lua (you can leave it in the readme, but an action just auto updates that bit)

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -694,6 +694,14 @@ list.hocon = {
   maintainers = { "@antosha417" },
 }
 
+list.hoon = {
+  install_info = {
+    url = "https://github.com/urbit-pilled/tree-sitter-hoon",
+    files = { "src/parser.c" },
+  },
+  maintainers = { "@urbit-pilled" },
```
Comment (amaanq) [inline lua/nvim-treesitter/parsers.lua:702]:
```suggestion
  maintainers = { "@urbit-pilled" },
  experimental = true,
```

File: queries/hoon/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "]"
+]  @punctuation.bracket
+
+[
+  (coreTerminator)
+  (seriesTerminator)
+] @punctuation.delimiter
+
+
+(rune) @keyword
```
Comment (amaanq) [inline queries/hoon/highlights.scm:18]: from the looks of it, isn't this an operator (or punctuation.(delimiter|special)?

File: queries/hoon/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+(term) @constant
+
+(aura) @constant.builtin
+
+(Gap) @comment
+
+(boolean) @constant.builtin
+
+(date) @constant.builtin
```
Comment (amaanq) [inline queries/hoon/highlights.scm:28]: Maybe `@string.special`?

File: queries/hoon/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+(mold) @constant.builtin
```
Comment (amaanq) [inline queries/hoon/highlights.scm:29]:
```suggestion
(mold) @operator
```
Or `@punctuation.special`

File: queries/hoon/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+(specialIndex) @constant.builtin
```
Comment (amaanq) [inline queries/hoon/highlights.scm:30]: Maybe `@number.builtin`? Need more context on this one

File: queries/hoon/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+(lark) @operator
+(fullContext) @constant.builtin
```
Comment (amaanq) [inline queries/hoon/highlights.scm:32]:
```suggestion
(fullContext) @punctuation.delimiter
```

**Review by amaanq — CHANGES_REQUESTED** [review]

File: queries/hoon/highlights.scm (hunk 2 above)
Comment (urbit-pilled) [reply queries/hoon/highlights.scm:18]: correct, a rune should be an operator

File: queries/hoon/highlights.scm (hunk 4 above)
Comment (urbit-pilled) [reply queries/hoon/highlights.scm:29]: Here's the documentation for [molds](https://developers.urbit.org/reference/hoon/irregular#trivial-molds), it's a symbolic alias for a type, (noun, atom cell, boolean, null). So I think this should be `@symbol`

File: queries/hoon/highlights.scm (hunk 5 above)
Comment (urbit-pilled) [reply queries/hoon/highlights.scm:30]: The specialIndex rule represents 3 different ways to index an array:
- `&n` nth element
- `|n` tail of list after nth element
- `+n` slot n (tree-addressing)

all of which are converted to an numerical index, so I think `@number.builtin` would be suitable.
https://developers.urbit.org/reference/hoon/irregular#list-addressing

File: queries/hoon/highlights.scm (hunk 6 above)
Comment (urbit-pilled) [reply queries/hoon/highlights.scm:32]: This one is a similar to mold, it is a symbol alias. So I think `@symbol` would be suitable.

File: queries/hoon/highlights.scm (hunk 3 above)
Comment (urbit-pilled) [reply queries/hoon/highlights.scm:28]: right

**Review by amaanq — APPROVED** [review]

**Pull request merged**